packages/dashboard/src/customization/propertiesSections/propertiesAndAlarmsSettings/styledPropertyComponent.tsx

```diff
@@ -88,30 +98,68 @@ export type StyledPropertyComponentProps = {
   updateStyle: (newStyles: object) => void;
   assetSummary: AssetSummary;
   property: StyledAssetPropertyQuery;
+  onHideAssetQuery: () => void;
   onDeleteAssetQuery?: () => void;
   colorable: boolean;
+  isPropertyVisible: boolean;
 };
 
 export const StyledPropertyComponent: FC<StyledPropertyComponentProps> = ({
   assetSummary,
   property,
   updateStyle,
+  onHideAssetQuery,
   onDeleteAssetQuery,
   colorable,
+  isPropertyVisible,
 }) => {
   const { display, label } = getPropertyDisplay(property.propertyId, assetSummary);
+  const [onMouseOver, setOnMouseOver] = useState(false);
+  const tooltipStyle = {
+    fontSize: spaceScaledM,
+    color: colorBackgroundHomeHeader,
+    backgroundColor: colorBackgroundLayoutMain,
+    padding: `${spaceStaticXs} ${spaceStaticXs}`,
+    borderRadius: spaceStaticXs,
+    borderWidth: spaceScaledXxxs,
+    borderColor: colorBorderButtonNormalDisabled,
+    boxShadow: `${spaceScaledXxxs} ${spaceScaledXxxs} ${spaceScaledXxxs} ${colorBorderButtonNormalDisabled}`,
+  };
 
   const resetStyles = (styleToReset: object) => {
     updateStyle(styleToReset); // as we add more sections, reset style values here
   };
 
+  const onToggleAssetQuery: NonCancelableEventHandler<ClickDetail> = (e) => {
+    e.stopPropagation();
+    onHideAssetQuery();
+  };
+
+  const handleMouseEnter = () => {
+    setOnMouseOver(true);
+  };
+
+  const handleMouseLeave = () => {
+    setOnMouseOver(false);
+  };
+
   const YAxisHeader = (
     <SpaceBetween size='s' direction='horizontal'>
       {colorable && display === 'property' && (
         <ColorPicker color={property.color || ''} updateColor={(newColor) => updateStyle({ color: newColor })} />
       )}
       <div style={{ fontWeight: 'normal' }}>{label}</div>
-      {/* <Button onClick={() => {}} variant='icon' iconSvg={StatusEyeVisible} /> TODO */}
+      <div className='property-display-toggle' onMouseEnter={handleMouseEnter} onMouseLeave={handleMouseLeave}>
+        <Button
+          onClick={onToggleAssetQuery}
+          variant='icon'
+          iconSvg={(!onMouseOver ? isPropertyVisible : !isPropertyVisible) ? StatusEyeVisible : StatusEyeHidden}
+        />
+        <span className='tooltiptext' style={tooltipStyle}>
+          {onMouseOver ? (isPropertyVisible ? 'hide' : 'unhide') : !isPropertyVisible ? 'hide' : 'unhide'}
```

Comment:
if the tooltip wont be visible once mouse leaves then why not just do 
`onMouseOver && (isPropertyVisible ? 'hide' : 'unhide')` ?

Reply:
done